app/lib/features/chat/widgets/custom_input.dart

```diff
@@ -534,6 +444,84 @@ class _CustomChatInputState extends ConsumerState<CustomChatInput> {
     );
   }
 
+  void onPressEditMessage(String roomId, String? currentMessageId) {
+    final emojiRowVisible = ref.read(
+      chatInputProvider(roomId).select((ci) {
+        return ci.emojiRowVisible;
+      }),
+    );
+    final inputNotifier = ref.read(chatInputProvider(roomId).notifier);
+    if (emojiRowVisible) {
+      inputNotifier.setCurrentMessageId(null);
+      inputNotifier.emojiRowVisible(false);
+    }
+
+    inputNotifier.showEditView(true);
+    final message =
+        ref.read(chatStateProvider(widget.convo)).messages.firstWhere(
+              (element) => element.id == currentMessageId,
+            );
+    inputNotifier.setEditMessage(message);
+    if (message is TextMessage) {
```

Comment:
I was referring to exporting only this part of the function as it could be testable without side-effects (like without the `ref.read`) by just having input->output unittests...

but it's fine for now. let's strive ahaed.

Reply:
@gnunicorn 
Added uni test cased for edit message with user mentions feature. :)

Also uploaded video in this PR description which shows test case execution.

Reply:
@gnunicorn 
I have separated main logic of message parsing and added unit test for case the same.